rollout/trafficrouting.go

```diff
@@ -17,152 +17,177 @@ import (
 
-		// Checks for experiment step
-		// If current experiment exists, then create WeightDestinations for each experiment template
-		exStep := replicasetutil.GetCurrentExperimentStep(c.rollout)
-		if exStep != nil && c.currentEx != nil && c.currentEx.Status.Phase == v1alpha1.AnalysisPhaseRunning {
-			getTemplateWeight := func(name string) *int32 {
-				for _, tmpl := range exStep.Templates {
-					if tmpl.Name == name {
-						return tmpl.Weight
+			// Checks for experiment step
+			// If current experiment exists, then create WeightDestinations for each experiment template
+			exStep := replicasetutil.GetCurrentExperimentStep(c.rollout)
+			if exStep != nil && c.currentEx != nil && c.currentEx.Status.Phase == v1alpha1.AnalysisPhaseRunning {
+				getTemplateWeight := func(name string) *int32 {
+					for _, tmpl := range exStep.Templates {
+						if tmpl.Name == name {
+							return tmpl.Weight
+						}
 					}
+					return nil
+				}
+				for _, templateStatus := range c.currentEx.Status.TemplateStatuses {
+					templateWeight := getTemplateWeight(templateStatus.Name)
+					weightDestinations = append(weightDestinations, trafficrouting.WeightDestination{
+						ServiceName:     templateStatus.ServiceName,
+						PodTemplateHash: templateStatus.PodTemplateHash,
+						Weight:          *templateWeight,
+					})
 				}
-				return nil
-			}
-			for _, templateStatus := range c.currentEx.Status.TemplateStatuses {
-				templateWeight := getTemplateWeight(templateStatus.Name)
-				weightDestinations = append(weightDestinations, trafficrouting.WeightDestination{
-					ServiceName:     templateStatus.ServiceName,
-					PodTemplateHash: templateStatus.PodTemplateHash,
-					Weight:          *templateWeight,
-				})
 			}
 		}
-	}
 
-	err = reconciler.SetWeight(desiredWeight, weightDestinations...)
-	if err != nil {
-		c.recorder.Warnf(c.rollout, record.EventOptions{EventReason: "TrafficRoutingError"}, err.Error())
-		return err
-	}
-
-	// If we are in the middle of an update at a setWeight step, also perform weight verification.
-	// Note that we don't do this every reconciliation because weight verification typically involves
-	// API calls to the cloud provider which could incur rate limiting
-	shouldVerifyWeight := c.rollout.Status.StableRS != "" &&
-		c.rollout.Status.CurrentPodHash != c.rollout.Status.StableRS &&
-		currentStep != nil && currentStep.SetWeight != nil
-
-	if shouldVerifyWeight {
-		weightVerified, err := reconciler.VerifyWeight(desiredWeight, weightDestinations...)
+		err = reconciler.SetWeight(desiredWeight, weightDestinations...)
 		if err != nil {
+			c.recorder.Warnf(c.rollout, record.EventOptions{EventReason: "TrafficRoutingError"}, err.Error())
 			return err
 		}
-		if !weightVerified {
-			c.log.Infof("Desired weight (stepIdx: %d) %d not yet verified", *index, desiredWeight)
-			c.enqueueRolloutAfter(c.rollout, 10*time.Second)
-		} else {
-			c.log.Infof("Desired weight (stepIdx: %d) %d verified", *index, desiredWeight)
+
```

Comment:
same here.

Reply:
this is some visual effect of github diff.
open the file in ur IDE, you won't see any indentation issue.
Line 97 adds a loop, where we have to iterate over the list of trafficReconcilers.
indent is right